File: src/StockMarket.Domain/ScanResult.cs
using SharedKernel;
using StockMarket.Domain.Enums;
using System;

namespace StockMarket.Domain
{
    public class ScanResult : EntityBase
    {
        public int ScripId { get; set; }
        public int? BatchId { get; set; }
        public DateTime Date { get; set; }
        public decimal Price { get; set; }
        public OrderType ScanType { get; set; }
        public ScanResultStatus? Status { get; set; }
        public DateTime? LastModified { get; set; }
        public int Rank { get; set; }

        public Scrip Scrip { get; set; }
        public Batch Batch { get; set; }
    }
}

File: src/StockMarketWebClient/src/app/shared/models/ScanResult.ts
import { Scrip } from './Scrip';

export interface ScanResult {
    id: number;
    isDirty: boolean;
    scripId: string;
    batchId: number;
    date: Date;
    price: number;
    scanType: string;
    status: string;
    lastModified: Date;
    rank: number;

    scrip: Scrip
}